src/routes/auth/index.ts

```diff
@@ -1,142 +1,288 @@
+import { createRoute, OpenAPIHono } from '@hono/zod-openapi';
 import sgMail from '@sendgrid/mail';
 import { addHours } from 'date-fns';
-import { type Context, Hono } from 'hono';
+import { Hono } from 'hono';
 import { generateEmailHtml } from '../../email-templates/confirm-email.ts';
+import { authorizeVolunteer } from '../../middleware/createMiddleware.ts';
 import type { SessionData } from '../../types/data/session';
 import { hashPassword, validatePassword } from '../../utils/password-hashing.ts';
-import { StatusCodes, type StatusResponse } from '../../utils/responses.ts';
-import { insertProfile } from '../profile/data.ts';
+import { type HeartbeatResponse, HeartbeatResponseSchema, StatusCodes, type StatusResponse, statusResponseFormatter, StatusResponseSchema } from '../../utils/responses.ts';
+import { getProfileById, insertProfile } from '../profile/data.ts';
 import { type CreateProfileRequestBody, CreateProfileSchema } from '../profile/validation.ts';
 import { activateProfile, checkEmail, getLoginInfo } from './data.ts';
-import { type SignInData, SignInSchema } from './validate.ts';
-
+import { ActivateSchema, type SignInData, SignInSchema, SignUpSchema } from './validate.ts';
 export const authRoutes = new Hono();
 
-authRoutes.post('/sign-up', async (context: Context<EnvironmentBindings>) => {
-	let parsedBody: CreateProfileRequestBody;
+// Public Routes (Post, Get)
+export const publicAuthRoutes = new OpenAPIHono<EnvironmentBindings>({
+	defaultHook: statusResponseFormatter
+});
 
-	try {
-		const body = await context.req.json();
-		parsedBody = CreateProfileSchema.parse(body);
-	} catch (error) {
-		if (error instanceof SyntaxError) {
-			return context.json<StatusResponse>({ message: `Invalid JSON format: ${error.message}` }, StatusCodes.BAD_REQUEST);
+publicAuthRoutes.openapi(
+	createRoute({
+		method: 'post',
+		path: '/',
+		operationId: 'signUp',
+		description: 'Sign up to community hub account',
+		tags: ['Sign-up'],
+		request: {
+			body: { content: { 'application/json': { schema: SignUpSchema } }, required: true }
+		},
+		responses: {
+			[StatusCodes.BAD_REQUEST]: {
+				description: 'Invalid JSON format:',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.CONFLICT]: {
+				description: 'Duplicate email',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			},
+			[StatusCodes.OKAY]: {
+				description: 'Created a new profile and sent an email for confirmation',
+				content: { 'application/json': { schema: StatusResponseSchema } }
+			}
 		}
-		throw error;
-	}
+	}),
+	async (context) => {
+		let parsedBody: CreateProfileRequestBody;
 
-	const emailExists = await checkEmail(context.env.database, parsedBody.email);
-	if (emailExists) {
-		return context.json<StatusResponse>({ message: 'Duplicate email' }, StatusCodes.CONFLICT);
-	}
+		try {
+			const body = await context.req.json();
+			parsedBody = CreateProfileSchema.parse(body);
```

Comment:
suggestion: No need to parse the body here. Let Open API handle that based on the schema you pass on line 29 of this file

Reply:
```const parsedBody  = await context.valid.json()``` ???

Reply:
Intdocing the suggested code introduces a chain of new errors.